src/libs/bir_exp_to_wordsLib.sml

```diff
@@ -321,7 +321,8 @@ struct
                 val offset = Arbnumcore.- (n, Arbnumcore.one);
                 val offset_w = mk_word (offset, size_of base_addr_w);
                 val addr = ``^base_addr_w + ^offset_w``;
-                val load_tm = ``^addr :> ^mem_w``;
+                (* this may need to be refined to: case FLOOKUP .. .. of SOME x => x | NONE => 0w *)
```

Comment:
Since you wrote this comment, I wonder why you didn't do it. The semantic is that the value is `0w` if the address isn't mapped, so only using `FAPPLY` doesn't respect the semantic? Don't we risk soundness issue with the current code?

Reply:
FAPPLY is undefined for unmapped values. This function is for SMT solver interaction and we just assume fully defined maps for now.

Reply:
> FAPPLY is undefined for unmapped values

Oh right, so it's sound, but not complete. I am right?

Reply:
Under the assumption that we are dealing with fully defined maps it should be both if I don't miss anything now. In this case our semantics is equal to FAPPLY.

Reply:
Yes, under this assumption I think you're right. I was saying without this assumption.

I will mark this as resolved now that we are aware of this :)